core/templates/dev/head/domain/exploration/StatesObjectFactory.js

```diff
@@ -18,18 +18,112 @@
  */
 
 oppia.factory('StatesObjectFactory', [
-  'StateObjectFactory', function(StateObjectFactory) {
-    var createFromBackendDict = function(statesBackendDict) {
+  'StateObjectFactory', 'newStateTemplateService', 'INTERACTION_SPECS',
+  function(StateObjectFactory, newStateTemplateService, INTERACTION_SPECS) {
+    var States = function(states) {
+      var states = states;
```

Comment:
Ah, my bad. I'm not on top of my game here -- you're right about the non-singletonic nature of this. I like your solution of this._states, that's very nice, and we should adopt that for the others later (i.e. this._privateProperty and add getters/setters).